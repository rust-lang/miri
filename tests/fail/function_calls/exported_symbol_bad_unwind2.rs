//@revisions: extern_block definition both
#![feature(rustc_attrs, c_unwind)]

#[cfg_attr(any(definition, both), rustc_allocator_nounwind)]
#[no_mangle]
extern "C-unwind" fn nounwind() {
    //[definition]~^ ERROR: abnormal termination: the program aborted execution
    //[both]~^^ ERROR: abnormal termination: the program aborted execution
    panic!();
}

fn main() {
    extern "C-unwind" {
        #[cfg_attr(any(extern_block, both), rustc_allocator_nounwind)]
        fn nounwind();
    }
    unsafe { nounwind() }
    //[extern_block]~^ ERROR: unwinding past a stack frame that does not allow unwinding
}
