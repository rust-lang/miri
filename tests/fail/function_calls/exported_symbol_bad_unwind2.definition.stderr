thread 'main' panicked at 'explicit panic', $DIR/exported_symbol_bad_unwind2.rs:LL:CC
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: abnormal termination: the program aborted execution
  --> $DIR/exported_symbol_bad_unwind2.rs:LL:CC
   |
LL | / extern "C-unwind" fn nounwind() {
LL | |
LL | |
LL | |     panic!();
LL | | }
   | |_^ the program aborted execution
   |
   = note: inside `nounwind` at $DIR/exported_symbol_bad_unwind2.rs:LL:CC
note: inside `main` at $DIR/exported_symbol_bad_unwind2.rs:LL:CC
  --> $DIR/exported_symbol_bad_unwind2.rs:LL:CC
   |
LL |     unsafe { nounwind() }
   |              ^^^^^^^^^^

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

